tests/testsuite/clean.rs

```diff
@@ -96,26 +97,66 @@ fn clean_multiple_packages_in_glob_char_path() {
         .build();
     let foo_path = &p.build_dir().join("debug").join("deps");
 
+    #[cfg(not(target_env = "msvc"))]
+    let file_glob = "foo-*";
+
+    #[cfg(target_env = "msvc")]
+    let file_glob = "foo.pdb";
+
     // Assert that build artifacts are produced
     p.cargo("build").run();
-    assert_ne!(get_build_artifacts(foo_path).len(), 0);
+    assert_ne!(get_build_artifacts(foo_path, file_glob).len(), 0);
 
     // Assert that build artifacts are destroyed
     p.cargo("clean -p foo").run();
-    assert_eq!(get_build_artifacts(foo_path).len(), 0);
+    assert_eq!(get_build_artifacts(foo_path, file_glob).len(), 0);
 }
 
-fn get_build_artifacts(path: &PathBuf) -> Vec<Result<PathBuf, GlobError>> {
-    let pattern = path.to_str().expect("expected utf-8 path");
-    let pattern = glob::Pattern::escape(pattern);
+#[cargo_test]
+fn clean_p_only_cleans_specified_package() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [workspace]
+                members = [
+                    "foo",
+                    "foo_core",
+                ]
+            "#,
+        )
+        .file("foo/Cargo.toml", &basic_lib_manifest("foo"))
+        .file("foo/src/lib.rs", "//! foo")
+        .file("foo_core/Cargo.toml", &basic_lib_manifest("foo_core"))
+        .file("foo_core/src/lib.rs", "//! foo_core")
+        .build();
 
-    #[cfg(not(target_env = "msvc"))]
-    const FILE: &str = "foo-*";
+    let deps_path = &p.build_dir().join("debug").join("deps");
```

Comment:
I feel like this is an incorrect path. This PR handles fingerprints and build script artifacts, which should reside in `target/debug/.figerprint` and `target/debug/build` respectively.

Reply:
You're right, my mistake